src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/TypeResolver.Ecma.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+
+        //
+        // Main routine to resolve a typeDefinition.
+        //
+        internal static RuntimeTypeInfo ResolveTypeDefinition(this TypeDefinitionHandle typeDefinitionHandle, MetadataReader reader)
+        {
+            return typeDefinitionHandle.GetNamedType(reader);
+        }
+
+        //
+        // Main routine to parse a metadata type specification signature.
+        //
+        private static RuntimeTypeInfo TryResolveTypeSignature(this TypeSpecificationHandle typeSpecHandle, MetadataReader reader, TypeContext typeContext, ref Exception exception)
+        {
+            TypeSpecification typeSpec = reader.GetTypeSpecification(typeSpecHandle);
+            ReflectionTypeProvider refTypeProvider = new ReflectionTypeProvider(false);
+            RuntimeTypeInfo result = typeSpec.DecodeSignature<RuntimeTypeInfo, TypeContext>(refTypeProvider, typeContext);
+            exception = refTypeProvider.ExceptionResult;
+            return result;
+        }
+
+        private static string GetFullyQualifiedTypeName(this TypeReference typeReference, MetadataReader reader)
+        {
+            string fullName = reader.GetString(typeReference.Name);
+
+            if (!typeReference.Namespace.IsNil)
+                fullName = reader.GetString(typeReference.Namespace) + "." + fullName;
+
+            return fullName;
+        }
+        private static RuntimeTypeInfo TryResolveTypeByName(MetadataReader reader, string fullName, ref Exception exception)
+        {
+            RuntimeAssembly assembly = EcmaFormatRuntimeAssembly.GetRuntimeAssembly(reader);
+            return assembly.GetTypeCore(fullName, false);
+        }
+
+        //
+        // Main routine to resolve a typeReference.
+        //
+        private static RuntimeTypeInfo TryResolveTypeReference(this TypeReferenceHandle typeReferenceHandle, MetadataReader reader, ref Exception exception)
+        {
+            TypeReference typeReference = reader.GetTypeReference(typeReferenceHandle);
+            EntityHandle resolutionScope = typeReference.ResolutionScope;
+            if (resolutionScope.IsNil)
+            {
+                // Search for an entry in the exported type table for this type
+                return TryResolveTypeByName(reader, typeReference.GetFullyQualifiedTypeName(reader), ref exception);
+            }
+
+            switch (resolutionScope.Kind)
+            {
+                case HandleKind.ModuleReference:
+                    // multi-module assemblies are not supported by this runtime
+                    exception = new PlatformNotSupportedException();
+                    return null;
+
+                case HandleKind.TypeReference:
+                    // This is a nested type. Find the enclosing type, then search for a matching nested type
+                    RuntimeTypeInfo enclosingType = ((TypeReferenceHandle)typeReference.ResolutionScope).TryResolveTypeReference(reader, ref exception);
```

Comment:
If you're going to introduce new local variables in a case branch, pls enclose in curly braces to limit their scope.